book/tests/end_to_end.rs

```diff
@@ -0,0 +1,174 @@
+    assert_eq!(
+        book_not_found.unwrap_err(),
+        Error::NotFound,
+        "Database should be empty"
+    );
+}
+
+#[tokio::test]
+async fn should_find_book() {
+    // Arrange
+    let (_, mut client) = utils::test_setup().await;
+
+    // Act
+    let book_not_found = client.get_book(context::current(), 1).await.unwrap();
+
+    // Assert
+    assert_eq!(
+        book_not_found.unwrap(),
+        test_data::book_1(),
+        "Database should be empty"
+    );
+}
+
+mod test_data {
+    use chrono::NaiveDate;
+
+    use book::models::*;
+
+    pub fn book_1() -> Book {
+        Book {
+            id: 1,
+            code_identifier: "SE001".to_string(),
+            isbn: "1234567891011".to_string(),
+            issn: None,
+            release_date: NaiveDate::from_ymd(1900, 1, 1),
+            subtitle: Some("Why you should learn it".to_string()),
+            title: "Rust Book".to_string(),
+            category: Category {
+                id: 1,
+                name: "STS".to_string(),
+            },
+            language: Language {
+                id: 1,
+                iso_code: "aar".to_string(),
+                name: "Afar".to_string(),
+            },
+            publisher: Publisher {
+                id: 1,
+                name: "Nostarch Press".to_string(),
+            },
+            series: None,
+            authors: vec![
+                Person {
+                    id: 1,
+                    first_name: "Markus".to_string(),
+                    last_name: "Merklinger".to_string(),
+                    isni: None,
+                    orcid: None,
+                    oclc: None,
+                },
+                Person {
+                    id: 2,
+                    first_name: "Maya".to_string(),
+                    last_name: "Alroy".to_string(),
+                    isni: None,
+                    orcid: None,
+                    oclc: None,
+                },
+            ],
+            subject_areas: vec![],
+        }
+    }
+}
+
+mod utils {
+    use std::io;
+    use std::net::SocketAddr;
+    use std::sync::Arc;
+
+    use diesel::prelude::*;
+    use tarpc::client;
+    use tokio_serde::formats::Json;
+    use uuid::Uuid;
+
+    use book::db::{self, DbConn, DbPool};
+    use book::rpc_server;
+    use book::service::BookServiceClient;
+
+    use super::*;
+
+    pub async fn test_setup() -> (DbConn, BookServiceClient) {
+        // database
+        let db_pool = db_setup();
+        let db_conn = db_pool.get().unwrap();
+        db_test_data(&db_conn).unwrap();
+        // rpc
+        let addr = spawn_rpc_server(db_pool).await;
+        let client = get_rpc_client(addr).await.unwrap();
+
+        (db_conn, client)
+    }
+
+    fn db_setup() -> DbPool {
+        // create dummy database
+        let tmp_conn = PgConnection::establish(DATABASE_URL).unwrap();
+        let db_name = Uuid::new_v4();
+        diesel::sql_query(&*format!("CREATE DATABASE \"{}\";", db_name))
```

Comment:
```suggestion
        diesel::sql_query(&format!("CREATE DATABASE \"{}\"", db_name))
```

Reply:
But then it is `&String` and not `&str` :P

Reply:
And I think I need the `;`, cause ther query should be `CREATE DATABASE "abcdef";`.

Reply:
Yes, but it also works with String and is easier to read. The `diesel::sql_query()` will automatically append the semicolon.